main/views.py

```diff
@@ -1,57 +1,137 @@
+import json
+import datetime
+
+
 from django.http import HttpResponse, JsonResponse
 from django.shortcuts import get_object_or_404, render
-
 from .models import *
 
 
 def index(request):
+    # i will remove matches and use results instead
     matches = Match.objects.all()
-    results = {}
+    results = []
     for match in matches:
         if match.player1_score > match.player2_score:
             winner = match.player1.name
             loser = match.player2.name
-            w = match.player1_score
-            l = match.player2_score
+            w_score = match.player1_score
+            l_score = match.player2_score
         else:
             winner = match.player2.name
             loser = match.player1.name
-            w = match.player2_score
-            l = match.player1_score
-        results[match.id]={"winner":winner, "loser":loser, "l":l, "w":w}
+            w_score = match.player2_score
+            l_score = match.player1_score
+        results.append({"winner": winner, "loser": loser, "l": l_score, "w": w_score})
     print results
     context = {
         'matches': matches,
         'results': results,
     }
     return render(request, 'main/index.html', context)
 
+
 def players(request):
     players = Player.objects.all()
     context = {
         'players': players,
     }
     return render(request, 'main/players.html', context)
 
+
 def fields(request):
     fields = Field.objects.all()
     context = {
         'fields': fields,
     }
     return render(request, 'main/fields.html', context)
 
+
 def player_detail(request, player_id):
     player = get_object_or_404(Player, pk=player_id)
     return render(request, 'main/player_detail.html', {'player': player})
 
+
 def api_players(request):
-    players = Player.objects.all()
-    data = []
-    for player in players:
-        data.append({"id": player.id, "name": player.name})
-    return JsonResponse({"players": data})
+    if request.method == "GET":
+        players = Player.objects.all()
+        data = []
+        for player in players:
+            data.append({"id": player.id, "name": player.name})
```

Comment:
These three lines could be accomplished in one using a dict comprehension - look into those and try to update the code here.

Reply:
Actually, here and in the other API endpoints as well.